src/AutoRest.CSharp/Mgmt/Generation/MgmtClientBaseWriter.cs

```diff
@@ -354,15 +520,8 @@ protected void WriteClientMethod(CodeWriter writer, ClientMethod clientMethod, s
                     {
                         writer.Append($"await ");
                     }
-
-                    var parameterNames = GetParametersName(restClientMethod, operationGroup, context);
                     writer.Append($"{restClientName ?? RestClientField}.{CreateMethodName(restClientMethod.Name, async)}(");
-                    // TODO -- we need to change this to BuildAndWriteParameters(writer, clientMethod) to make it be able to handle more cases
-                    // but directly replace the following logic by this function is causing issues
```

Comment:
@ArcturusZhang I've made some fix in BuildParameterMapping and ShouldPassThrough, then using BuildAndWriteParameters here looks fine from the test eaxmples and actually fixed some parameter value expressions in MgmtListMethods. Please let me know if you have any concerns.